packages/utils/src/lib/git.ts

```diff
@@ -38,17 +38,67 @@ export async function toGitPath(
 
 export function validateCommitData(
   commitData: CommitData | null,
-  options: { throwError?: boolean } = {},
+  options: { throwError?: true } = {},
 ): commitData is CommitData {
-  const { throwError = false } = options;
   if (!commitData) {
     const msg = 'no commit data available';
-    if (throwError) {
+    if (options?.throwError) {
       throw new Error(msg);
     } else {
+      // @TODO replace with ui().logger.warning
       console.warn(msg);
       return false;
     }
   }
   return true;
 }
+
+export async function guardAgainstLocalChanges(
+  git = simpleGit(),
+): Promise<void> {
+  const isClean = await git.status(['-s']).then(r => r.files.length === 0);
+  if (!isClean) {
+    throw new Error(
+      'Working directory needs to be clean before we you can proceed. Commit your local changes or stash them.',
+    );
+  }
+}
+
+export async function getCurrentBranchOrTag(
+  git = simpleGit(),
+): Promise<string> {
+  try {
+    const branch = await git.branch().then(r => r.current);
+    // eslint-disable-next-line unicorn/prefer-ternary
+    if (branch) {
+      return branch;
+    } else {
+      // If no current branch, try to get the tag
+      // @TODO use simple git
+      return await git
+        .raw(['describe', '--tags', '--exact-match'])
+        .then(out => out.trim());
+    }
+  } catch {
+    // Return a custom error message when something goes wrong
+    throw new Error('Could not get current tag or branch.');
+  }
+}
+
+export async function safeCheckout(
+  branchOrHash: string,
+  options: {
+    forceCleanStatus?: true;
+  } = {},
+  git = simpleGit(),
+): Promise<void> {
+  // git requires a clean history to check out a branch
+  if (options?.forceCleanStatus) {
+    await git.raw(['reset', '--hard']);
+    await git.clean(['f', 'd']);
```

Comment:
Maybe this can be done to delete the branch?

Reply:
I tried it. :(